### PR TITLE
Add ruff for linting, remove flake8, remove isort, remove pylint

### LINKED ISSUE #765: Switch to using ruff for Python linting and code formatting
### Problem Description
- We currently use `isort`, `flake8`, and `pydocstyle` for managing out python linting and code formatting.
- However, many libraries have switched to using `ruff`. It is a much faster linter/formatter and can replace the `isort`/`flake8`/`pydocstyle` libraries (among others)

### Expected behavior
- Use `ruff` as the standard linter/formatter

### PR DESCRIPTION
**Pull request by gsheni**
- Closes #765

### CODE REVIEW

File: rdt/__init__.py [hunk 1]
```diff
@@ -42,10 +38,13 @@ def get_demo(num_rows=5):
...
     """
     # Hard code first five rows
-    login_dates = pd.Series([
-        '2021-06-26', '2021-02-10', 'NAT', '2020-09-26', '2020-12-22'
-    ], dtype='datetime64[ns]')
-    email_optin = pd.Series([False, False, False, True, np.nan], dtype='object')
+    login_dates = pd.Series(
+        ['2021-06-26', '2021-02-10', 'NAT', '2020-09-26', '2020-12-22'],
+        dtype='datetime64[ns]',
+    )
```
Comment (amontanez24) [inline rdt/__init__.py:44]: anyway we can configure it to allow the paranthesis closing on an indented line like
```python
x = func(
    param1, param2, param3)
```

File: rdt/__init__.py [hunk 2]
```diff
@@ -67,33 +66,47 @@ def get_demo(num_rows=5):
...
+            ],
+            dtype='datetime64[ns]',
+        )
+        login_dates[np.random.random(size=num_rows) > 0.8] = np.datetime64(
+            'NaT'
+        )
 
         email_optin = pd.Series([True, False, np.nan], dtype='object').sample(
-            num_rows, replace=True)
-        credit_card = np.random.choice(['VISA', 'AMEX', np.nan, 'DISCOVER'], size=num_rows)
```
Comment (amontanez24) [inline rdt/__init__.py:78]: and to allow one liners up to 100 chars

File: rdt/hyper_transformer.py [hunk 3]
```diff
@@ -495,11 +543,19 @@ def update_transformers(self, column_name_to_transformer):
...
-            columns = column_name if isinstance(column_name, tuple) else (column_name,)
+            columns = (
+                column_name
+                if isinstance(column_name, tuple)
+                else (column_name,)
+            )
             for column in columns:
                 if transformer is not None:
                     col_sdtype = self.field_sdtypes.get(column)
-                    if col_sdtype and col_sdtype not in transformer.get_supported_sdtypes():
```
Comment (amontanez24) [inline rdt/hyper_transformer.py:502]: and this kind of if statement if it fits on one line

File: rdt/transformers/datetime.py [hunk 4]
```diff
@@ -49,9 +50,14 @@ class UnixTimestampEncoder(BaseTransformer):
     _min_value = None
     _max_value = None
 
-    def __init__(self, missing_value_replacement='mean', model_missing_values=None,
-                 datetime_format=None, missing_value_generation='random',
-                 enforce_min_max_values=False):
```
Comment (amontanez24) [inline rdt/transformers/datetime.py:54]: and this type of init

**Review by amontanez24 — COMMENTED** [review]
this is really cool! It would be nice to configure it not to expand as many statements but other than that it's good

File: rdt/__init__.py (hunk 1 above)
Comment (gsheni) [reply rdt/__init__.py:44]: Looking through [this page](https://docs.astral.sh/ruff/settings/#format_indent-style), I did not find any settings for this. ruff (which is replacing black) is meant to be:
> Black is the uncompromising Python code formatter. By using it, you agree to cede control over minutiae of hand-formatting. In return, Black gives you speed, determinism, and freedom from pycodestyle nagging about formatting. You will save time and mental energy for more important matters.

File: rdt/__init__.py (hunk 2 above)
Comment (gsheni) [reply rdt/__init__.py:78]: Changed to `line-length = 100`

File: rdt/transformers/datetime.py (hunk 4 above)
Comment (gsheni) [reply rdt/transformers/datetime.py:54]: Did not find anything to change this. See above comment

**Comment by gsheni** [comment]
Also added docstring formatting:
```
docstring-code-format = true
docstring-code-line-length = "dynamic"
```

File: Makefile [hunk 5]
```diff
@@ -80,14 +80,14 @@ install-develop: clean-build clean-pyc ## install the package in editable mode a
 # LINT TARGETS
 
 .PHONY: lint
-lint: ## check style with flake8 and isort
-	invoke lint
```
Comment (amontanez24) [inline Makefile:84]: since you updated the lint command in tasks.py you can just leave this

**Review by amontanez24 — COMMENTED** [review]
One small comment

File: Makefile (hunk 5 above)
Comment (gsheni) [reply Makefile:84]: Fixed:
https://github.com/sdv-dev/RDT/pull/809/commits/2c4a754ab473d822d529cd81624d07a3520a6bd9

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**